muxers/mplex/tests/async_write.rs

```diff
@@ -64,8 +64,8 @@ fn async_write() {
             .await
             .unwrap();
 
-        let mut buf = Vec::new();
-        outbound.read_to_end(&mut buf).await.unwrap();
+        let mut buf = vec![0u8; 11];
+        outbound.read_exact(&mut buf).await.unwrap();
```

Comment:
Would that not defeat the purpose of the unit test?

Above:

>     // Tests that `AsyncWrite::close` implies flush.

Not using `read_to_end` thus no longer tests the closing and flushing, no?

Reply:
Yeah you might be right. I wasn't too happy about these changes but extracted them as a separate commit so we can discuss it easier.

I suspect there might be a bug somewhere but I couldn't spot it so far.

Reply:
I fixed it! The important commit is https://github.com/libp2p/rust-libp2p/commit/cec117667b3e0a23b8acb3acbc342e087ef8df6b.

This means I can drop this commit from the PR history but that will require a force-push. I've noted it in the PR description as a task for the final clean-up once this PR is closer to being merged :)